scss/bootstrap/components/_checkbox-switch.scss

```diff
@@ -0,0 +1,104 @@
+.checkbox-switch {
+  padding-top: 0 !important;
+  position: relative;
+
+  input {
+    bottom: 0;
+    height: 0;
+    left: 0;
+    margin: 0;
+    right: 0;
+    top: 0;
+    visibility: hidden;
+    width: 0;
+
+    + span {
+      cursor: pointer;
+      padding-left: $crm-checkbox-switch-size-normal * 2;
+      user-select: none;
+    }
+
+    + span::before {
+      background-color: $crm-gray-matte;
+      border-radius: $crm-checkbox-switch-size-normal;
+      content: '';
+      display: inline-block;
+      height: $crm-checkbox-switch-size-normal;
+      left: 0;
+      position: absolute;
+      transition: background-color $crm-checkbox-switch-speed ease-out;
+      width: $crm-checkbox-switch-size-normal * 2;
+    }
+
+    + span::after {
+      background-clip: padding-box;
+      background-color: $crm-white;
+      border: solid transparent $crm-checkbox-switch-gap;
+      border-radius: $crm-checkbox-switch-size-normal;
+      content: '';
+      display: block;
+      height: $crm-checkbox-switch-size-normal;
+      left: 0;
+      position: absolute;
+      top: 0;
+      transition: margin-left ($crm-checkbox-switch-speed / 2) ease-in-out;
+      width: $crm-checkbox-switch-size-normal;
+    }
+  }
+
+  input:checked {
+    + span::after {
+      animation: popIn ease-in ($crm-checkbox-switch-speed * 1.5) normal;
+      background-clip: padding-box;
+      margin-left: $crm-checkbox-switch-size-normal;
+    }
+
+    + span::before {
+      background-color: $brand-primary;
+      transition: background-color $crm-checkbox-switch-speed ease-in;
+    }
+  }
+
+  input:not(:checked) {
+    + span::after {
+      animation: popOut ease-in ($crm-checkbox-switch-speed * 1.5) normal;
+    }
+  }
+
+  /* Context (colours) */
+
+  &.checkbox-switch-warning {
+    @include checkbox-switch-context($brand-warning);
+  }
+
+  &.checkbox-switch-danger {
+    @include checkbox-switch-context($brand-danger);
+  }
+
+  &.checkbox-switch-success {
+    @include checkbox-switch-context($brand-success);
+  }
+
+  &.checkbox-switch-info {
+    @include checkbox-switch-context($brand-info);
+  }
+
+  /* Availability */
+
+  [disabled] {
```

Comment:
what is `[disabled]`? Shouldn't this be `input[disabled]`

Reply:
You cannot have `input[disabled]` - the linter will yell:

```
 ❯ Running tasks for *.scss
   ✖ stylelint --fix
     → 88:20  ✖  Unexpected qualifying type selector   selector-no-qualifying-type
     git add
✖ stylelint --fix found some errors. Please fix them and try committing again.
```